### PR TITLE
Fetching workload label selector from service on cluster directly

### PR DESCRIPTION
**Pull request by videlov**
**Description**

Changes proposed in this pull request:

- Fetching workload label selector from service on cluster directly

**Related issue(s)**

https://github.com/kyma-project/api-gateway/issues/278

### CODE REVIEW

**Comment by videlov** [comment]
> Do you think that adding a integration scenario for this makes sense?

yes ofc, why not? generally if it is possible to test something that has specifics while running on a real cluster with integration test I would do it. Even thought, I would also make sure that calling endpoint works as well, which I did not test. Perhaps I should extend the test.

**Review by barchw — CHANGES_REQUESTED** [review]
The label selector needs to be applied both to Authorization Policy and Request Authentication

File: tests/integration/features/istio-jwt/istio_jwt.feature [hunk 1]
```diff
@@ -128,3 +128,11 @@ Feature: Exposing endpoints with Istio JWT authorization strategy
     And JwtTokenFromParams: Calling the "/ip" endpoint with a valid "JWT" token from default header should result in status between 400 and 403
     And JwtTokenFromParams: Calling the "/ip" endpoint with a valid "JWT" token from parameter "jwt_token" should result in status between 200 and 299
     And JwtTokenFromParams: Teardown httpbin service
+
+  Scenario: Calling a helloworld endpoint with custom label selector service
+    Given CustomLabelSelector: There is a helloworld service with custom label selector name "custom-name"
+    And CustomLabelSelector: There is an endpoint secured with JWT on path "/ip"
+    When CustomLabelSelector: The APIRule is applied
+    Then CustomLabelSelector: Calling the "/ip" endpoint without a token should result in status between 400 and 403
+    And CustomLabelSelector: Calling the "/ip" endpoint with a valid "JWT" token should result in status between 200 and 299
```
Comment (barchw) [inline tests/integration/features/istio-jwt/istio_jwt.feature:137]: Helloworld app doesn't have /ip endpoint
```suggestion
    And CustomLabelSelector: There is an endpoint secured with JWT on path "/hello"
    When CustomLabelSelector: The APIRule is applied
    Then CustomLabelSelector: Calling the "/hello" endpoint without a token should result in status between 400 and 403
    And CustomLabelSelector: Calling the "/hello" endpoint with a valid "JWT" token should result in status between 200 and 299
```

**Review by barchw — CHANGES_REQUESTED** [review]

File: internal/processing/istio/authorization_policy_processor.go [hunk 2]
```diff
@@ -12,10 +13,13 @@ import (
...
 	"istio.io/api/security/v1beta1"
 	securityv1beta1 "istio.io/client-go/pkg/apis/security/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
-	audienceKey string = "request.auth.claims[aud]"
+	audienceKey          string = "request.auth.claims[aud]"
+	appSelectorLabel     string = "app"
+	serviceSelectorLabel string = "service"
```
Comment (barchw) [inline internal/processing/istio/authorization_policy_processor.go:22]: Both of those new variables are unused
```suggestion
```

**Review by barchw — CHANGES_REQUESTED** [review]
Looks good either than the small linting issue, can be merged after we figure out how to propagate RBAC without updating Kyma resources

**Review by barchw — APPROVED** [review]

**Pull request merged**